flink-mesos/src/test/scala/org/apache/flink/mesos/scheduler/LaunchCoordinatorTest.scala

```diff
@@ -0,0 +1,439 @@
+    val taskUnassigner = mock(classOf[Action2[String, String]])
+    when[Action2[String, String]](optimizer.getTaskUnAssigner).thenReturn(taskUnassigner)
+    optimizer
+  }
+
+  /**
+    * Create a task scheduler builder.
+    */
+  def taskSchedulerBuilder(optimizer: TaskScheduler) = new TaskSchedulerBuilder {
+    var leaseRejectAction: Action1[VirtualMachineLease] = null
+    override def withLeaseRejectAction(action: Action1[VirtualMachineLease]): TaskSchedulerBuilder = {
+      leaseRejectAction = action
+      this
+    }
+    override def build(): TaskScheduler = optimizer
+  }
+
+  /**
+    * Process a call to scheduleOnce with the given function.
+    */
+  def scheduleOnce(f: (Seq[TaskRequest],Seq[VirtualMachineLease]) => SchedulingResult) = {
+    new Answer[SchedulingResult] {
+      override def answer(invocationOnMock: InvocationOnMock): SchedulingResult = {
+        val args = invocationOnMock.getArguments
+        val requests = args(0).asInstanceOf[java.util.List[TaskRequest]]
+        val newLeases = args(1).asInstanceOf[java.util.List[VirtualMachineLease]]
+        f(requests.asScala, newLeases.asScala)
+      }
+    }
+  }
+
+  /**
+    * The context fixture.
+    */
+  class Context {
+    val optimizer = taskScheduler()
+    val optimizerBuilder = taskSchedulerBuilder(optimizer)
+    val schedulerDriver = mock(classOf[SchedulerDriver])
+    val trace = Mockito.inOrder(schedulerDriver)
+    val fsm = TestFSMRef(new LaunchCoordinator(testActor, config, schedulerDriver, optimizerBuilder))
+
+    val framework = randomFramework
+    val task1 = randomTask
+    val task2 = randomTask
+    val task3 = randomTask
+
+    val slave1 = {
+      val slave = randomSlave
+      (slave._1, slave._2, randomOffer(framework, slave), randomOffer(framework, slave), randomOffer(framework, slave))
+    }
+
+    val slave2 = {
+      val slave = randomSlave
+      (slave._1, slave._2, randomOffer(framework, slave), randomOffer(framework, slave), randomOffer(framework, slave))
+    }
+  }
+
+  def inState = afterWord("in state")
+  def handle = afterWord("handle")
+
+  def handlesAssignments(state: TaskState) = {
+    "Unassign" which {
+      s"stays in $state with updated optimizer state" in new Context {
+        optimizer.getTaskAssigner.call(task1._2.taskRequest, slave1._2)
+        fsm.setState(state)
+        fsm ! Unassign(task1._1, slave1._2)
+        verify(optimizer.getTaskUnAssigner).call(task1._1.getValue, slave1._2)
+        fsm.stateName should be (state)
+      }
+    }
+    "Assign" which {
+      s"stays in $state with updated optimizer state" in new Context {
+        fsm.setState(state)
+        fsm ! Assign(Seq(new FlinkTuple2(task1._2.taskRequest, slave1._2)).asJava)
+        verify(optimizer.getTaskAssigner).call(MM.any(), MM.any())
+        fsm.stateName should be (state)
+      }
+    }
+  }
+
+  "The LaunchCoordinator" when inState {
+
+    "Suspended" should handle {
```

Comment:
Correct, the text alone doesn't set the initial state.   The test bodies call `fsm.setState` to establish the initial state (along with corresponding state data).

I found the wordspec DSL to be a good fit for Akka FSM testing, because it can be made to closely match the FSM concept of being in some state and handling specific events.  The wordspec test output is delightful, with a nice tree structure in the IDE and with nice sentences in the console.